text/0756-helper-default-manager.md

```diff
@@ -156,6 +176,57 @@ setHelperManager(() => DEFAULT_HELPER_MANAGER, Function.prototype);
  - to register this helper manager, it should occur during app boot so developers do not need to import anything to
    trigger the `setHelperManager` call
 
+**Update 2022-01-13, during implementation**
+During the implementation of this feature, https://github.com/glimmerjs/glimmer-vm/pull/1348
+an issue came up where this RFC doesn't _explicitly_ say what to do with undefined arguments and default arguments.
+
+For example, it was originally assumed that the _last arg_ passed to the helper-function would always be the
+named arguments -- "options-style". This is problematic for positional-only APIs where someone may want to do
+```hbs
+{{myHelper 1 2}}
+or
+{{myHelper 1}}
+```
+where `myHelper` is defined as
+```js
+function myHelper(a, b = 3) {
+  return a * b;
+}
+```
+_with only_ positional arguments, this works as expected, but if named arguments get involved,
+```hbs
+{{myHelper 1 optionA=true}}
+```
+then the named args are passed in place of `b`, which is expected to be a number.
+
+As a result, of this constraint, and somewhat based on the assumption that optional positional arguments
+are not commonly used in addition to named arguments,
+this RFC proposes a community constraint / convention that if named arguments are used, the positional
+arguments may not be optional.
+Behaviorally, there is nothing wrong with this, as a developer may want to do arity checks for maximum flexibility
+in their API.
+
+For example:
+```js
+function myHelper(...args) {
+  let [ maybeOptions, ...reversed ] = args.reverse();
+  let options = {};
+  let positional = [];
+
+  // this is dependent on what myHelper accepts
+  if (typeof maybeOptions === 'object') {
+    // potentially assume named arguments
+    options = maybeOptions;
+    positional = reversed.reverse();
+  } else {
+    // maybe a number was passed
+    positional = args;
+  }
+
+  // ...
+}
```

Comment:
You can adjust this accordingly based on the changes above. I think the mainly we should talk (in sentences, not paragraphs) about:

* The drawbacks
  * The consistency issue
  * Hazards when mixing defaulted positional arguments with named
  * This needs to be preserved if we add a spread operator/operation in handlebars
* The bottomline
  * Defaulted positional arguments are very common in functions, especially ones you import from libraries like loadash
  * Mixing defaulted positional arguments with options is relatively uncommon because they have the same hazard in plain JavaScript
  * Therefore this is probably worth it

Reply:
Updated! thanks!